test/RelayClient.test.ts

```diff
@@ -1176,5 +1129,403 @@ describe('RelayClient', function () {
+        const callStub = sandbox.stub().throws(new Error());
+        const deployVerifierStub = {
+          callStatic: {
+            verifyRelayedCall: callStub
+          }
+        } as unknown as DeployVerifier;
+        sandbox.stub(DeployVerifier__factory, 'connect').returns(deployVerifierStub);
+        const verifyWithVerifiers = relayClient._verifyWithVerifiers(FAKE_DEPLOY_TRANSACTION_REQUEST);
+
+        expect(callStub).to.be.calledOnce;
+        await expect(verifyWithVerifiers).to.be.rejected;
+      });
+
+    });
+
+    describe('_verifyWithRelayHub', function () {
+
+      let relayWorkerAddress: string;
+      let fakeMaxPossibleGas: BigNumber;
+      let relayHubStub: Sinon.SinonStubbedInstance<RelayHub>;
+
+      beforeEach(function () {
+        relayWorkerAddress = createRandomAddress();
+        fakeMaxPossibleGas = createRandomBigNumber(10000);
+      });
+
+      it('should allow if enveloping relay request will succeed', async function () {
+        const callStub = sandbox.stub().returns(undefined);
+        relayHubStub = {
+          callStatic: {
+            relayCall: callStub
+          }
+        } as unknown as typeof relayHubStub;
+        sandbox.stub(RelayHub__factory, 'connect').returns(relayHubStub);
+        const verifyWithRelayHub = relayClient._verifyWithRelayHub(relayWorkerAddress, FAKE_RELAY_TRANSACTION_REQUEST, fakeMaxPossibleGas);
+
+        expect(callStub).to.be.calledOnce;
+        await expect(verifyWithRelayHub).to.be.fulfilled;
+      });
+
+      it('should allow if enveloping deploy request will succeed', async function () {
+        const callStub = sandbox.stub().returns(undefined);
+        relayHubStub = {
+          callStatic: {
+            deployCall: callStub
+          }
+        } as unknown as typeof relayHubStub;
+        sandbox.stub(RelayHub__factory, 'connect').returns(relayHubStub);
+        const verifyWithRelayHub = relayClient._verifyWithRelayHub(relayWorkerAddress, FAKE_DEPLOY_TRANSACTION_REQUEST, fakeMaxPossibleGas);
+
+        expect(callStub).to.be.calledOnce;
+        await expect(verifyWithRelayHub).to.be.fulfilled;
+      });
+
+      it('should throw if enveloping relay request will fail', async function () {
+        const callStub = sandbox.stub().throws(new Error());
+        relayHubStub = {
+          callStatic: {
+            relayCall: callStub
+          }
+        } as unknown as typeof relayHubStub;
+        sandbox.stub(RelayHub__factory, 'connect').returns(relayHubStub);
+        const verifyWithRelayHub = relayClient._verifyWithRelayHub(relayWorkerAddress, FAKE_RELAY_TRANSACTION_REQUEST, fakeMaxPossibleGas);
+
+        expect(callStub).to.be.calledOnce;
+        await expect(verifyWithRelayHub).to.be.rejected;
+      });
+
+      it('should throw if enveloping deploy request will fail', async function () {
+        const callStub = sandbox.stub().throws(new Error());
```

Comment:
In some testing I throw error without description or dummy description since there are multiple scenarios possibles that were covered on their specific test

Reply:
in that case you should cover each scenario, or pick one, I'd say, otherwise this may be a false negative.

Reply:
👆🏽

Reply:
Ill cover each scenario

Reply:
updated